documentation/modules/security/ref-certificates-and-secrets.adoc

```diff
@@ -0,0 +1,151 @@
+Using private keys and certificates in PEM format means that users have to get them from the Secrets, and generate a corresponding truststore (or keystore) to use in their Java applications.
+PKCS #12 storage provides a truststore (or keystore) that can be used directly.
+
+All keys are 2048 bits in size.
+
+CA certificate validity periods are expressed as a number of days after certificate generation.
+You can configure the validity period of:
+
+* Cluster CA certificates in `Kafka.spec.clusterCa.validityDays`
+* Client CA certificates in `Kafka.spec.clientsCa.validityDays`
+
+== PKCS #12 storage
+
+PKCS #12 defines an archive file format (`.p12`) for storing cryptography objects into a single file with password protection.
+You can use PKCS #12 to manage certificates and keys in one place.
+
+Each Secret contains fields specific to PKCS #12.
+
+* The `.p12` field contains the certificates and keys.
+* The `.password` field is the password that protects the archive.
+
+The password is encoded  with base64.
+
+== Cluster CA Secrets
+
+.Cluster CA `Secrets` managed by the Cluster Operator in _<cluster>_
+[cols="3,3,4", options="header"]
+|===
+|Secret name
+|Field within Secret
+|Description
+
+|`_<cluster>_-cluster-ca`
+|`ca.key`
+|The current private key for the cluster CA.
+
+.3+|`_<cluster>_-cluster-ca-cert`
+|`ca.p12`
+|PKCS #12 archive file for storing certificates and keys.
+|`ca.password`
+|Password for protecting the  PKCS #12 archive file.
+|`ca.crt`
+|The current certificate for the cluster CA.
+
+
+.4+|`_<cluster>_-kafka-brokers`
+|`<cluster>-kafka-<num>.p12`
+|PKCS #12 archive file for storing certificates and keys.
+|`<cluster>-kafka-<num>.password`
+|Password for protecting the  PKCS #12 archive file.
+|`_<cluster>_-kafka-_<num>_.crt`
+|Certificate for Kafka broker pod _<num>_. Signed by a current or former cluster CA private key in `_<cluster>_-cluster-ca`.
+|`_<cluster>_-kafka-_<num>_.key`
+|Private key for Kafka broker pod _<num>_.
+
+.4+|`_<cluster>_-zookeeper-nodes`
+|`<cluster>-zookeeper-<num>.p12`
+|PKCS #12 archive file for storing certificates and keys.
+|`<cluster>-zookeeper-<num>.password`
+|Password for protecting the  PKCS #12 archive file.
+|`_<cluster>_-zookeeper-_<num>_.crt`
+|Certificate for ZooKeeper node _<num>_. Signed by a current or former cluster CA private key in `_<cluster>_-cluster-ca`.
+|`_<cluster>_-zookeeper-_<num>_.key`
+| Private key for ZooKeeper pod _<num>_.
+
+.4+|`_<cluster>_-entity-operator-certs`
+|`entity-operator_.p12`
+|PKCS #12 archive file for storing certificates and keys.
+|`entity-operator_.password`
+|Password for protecting the  PKCS #12 archive file.
+|`entity-operator_.crt`
+|Certificate for TLS communication between the Entity Operator and Kafka or ZooKeeper.
+Signed by a current or former cluster CA private key in `_<cluster>_-cluster-ca`.
+|`entity-operator.key`
+|Private key for TLS communication between the Entity Operator and Kafka or ZooKeeper
+
+|===
+
+
```

Comment:
Does the triple empty line have any special Asciidoc meaning to end the table or so? If not, we should probably delete the lines.